src/tests/java/picard/analysis/TheoreticalSensitivityTest.java

```diff
@@ -0,0 +1,241 @@
+import java.util.Arrays;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.File;
+import java.util.Scanner;
+
+/**
+ * Created by davidben on 5/18/15.
+ */
+public class TheoreticalSensitivityTest {
+
+    private final static File TEST_DIR = new File("testdata/picard/analysis/TheoreticalSensitivity/");
+    private final File depth = new File(TEST_DIR, "Solexa332667_DepthDist.histo");
+    private final File baseQ = new File(TEST_DIR, "Solexa332667_BaseQ.histo");
+
+    @Test
+    public void testRouletteWheel() throws Exception {
+
+        //test that a deterministic roulette wheel only gives one value
+        final double[] deterministicWeights = {0.0, 1.0, 0.0};
+        final TheoreticalSensitivity.RouletteWheel deterministicWheel = new TheoreticalSensitivity.RouletteWheel(deterministicWeights);
+        for (int n = 0; n < 10; n++) Assert.assertEquals(deterministicWheel.draw(), 1);
+
+        //test the sums of this deterministic wheel: a sum of n 1's equals n
+        final List<ArrayList<Integer>> deterministicSums = deterministicWheel.sampleCumulativeSums(10, 1);
+        for (int n = 0; n < 10; n++) Assert.assertEquals(deterministicSums.get(n).get(0), (Integer) n);
+    }
+
+    @Test
+    public void testProportionsAboveThresholds() throws Exception {
+        final List<ArrayList<Integer>> sums = new ArrayList<ArrayList<Integer>>();
+        sums.add(new ArrayList<Integer>(Arrays.asList(0,0,0)));
+        sums.add(new ArrayList<Integer>(Arrays.asList(10, 10)));
+        sums.add(new ArrayList<Integer>(Arrays.asList(5, 11, -2, 4)));
+        final List<Double> thresholds = Arrays.asList(-1.0, 1.0, 6.0);
+        Assert.assertEquals(sums.size(), 3);
+        Assert.assertEquals(thresholds.size(), 3);
+
+        final List<ArrayList<Double>> proportions = TheoreticalSensitivity.proportionsAboveThresholds(sums, thresholds);
+        Assert.assertEquals(proportions.size(), 3);
+
+        Assert.assertEquals(proportions.get(0).get(0), (double) 3/3);
+        Assert.assertEquals(proportions.get(0).get(1), (double) 0/3);
+        Assert.assertEquals(proportions.get(0).get(2), (double) 0/3);
+        Assert.assertEquals(proportions.get(1).get(0), (double) 2/2);
+        Assert.assertEquals(proportions.get(1).get(1), (double) 2/2);
+        Assert.assertEquals(proportions.get(1).get(2), (double) 2/2);
+        Assert.assertEquals(proportions.get(2).get(0), (double) 3/4);
+        Assert.assertEquals(proportions.get(2).get(1), (double) 3/4);
+        Assert.assertEquals(proportions.get(2).get(2), (double) 1/4);
+    }
+
+    @Test
+    public void testHetAltDepthDistribution() throws Exception {
+        final int N = 6;
+        final double p = 0.5;
+        final List<ArrayList<Double>> distribution = TheoreticalSensitivity.hetAltDepthDistribution(N);
+
+        for (int n = 0; n < N-1; n++) {
+            for (int m = 0; m <= n; m++) {
+                //java has no built-in binomial coefficient
```

Comment:
[Another of my mistakes] switch this comment to `//TODO: . . .` so that the IDE recognizes it.